app/models/user.rb

```diff
@@ -0,0 +1,27 @@
+class User < ActiveRecord::Base
+  has_many :nono_users
+  has_many :nonos, through: :nono_users
+  has_many :rsvps
+  has_many :parties, through: :rsvps
+
+  has_secure_password
+
+  validates :email, uniqueness: true, presence: true
+  validates_format_of :email, :with => /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i, :on => :create
```

Comment:
You might want to remove the on_create so that it validates on update as well as create.